File: backend/src/main/java/com/bookbook/domain/lendList/dto/LendListResponseDto.java
package com.bookbook.domain.lendList.dto;

import com.bookbook.domain.rent.entity.Rent;
import lombok.Builder;
import lombok.Getter;

import java.time.LocalDateTime;

@Getter
@Builder
public class LendListResponseDto {
    
    private Integer id;
    private Long lenderUserId;
    private String title;
    private String bookTitle;
    private String author;
    private String publisher;
    private String bookCondition;
    private String bookImage;
    private String address;
    private String rentStatus;
    private String borrowerNickname;
    private LocalDateTime createdDate;
    private LocalDateTime modifiedDate;
    private LocalDateTime returnDate;
    private boolean hasReview;

    public static LendListResponseDto from(Rent rent, String borrowerNickname, LocalDateTime returnDate, boolean hasReview) {
        return LendListResponseDto.builder()
                .id(rent.getId())
                .lenderUserId(rent.getLenderUserId())
                .title(rent.getTitle())
                .bookTitle(rent.getBookTitle())
                .author(rent.getAuthor())
                .publisher(rent.getPublisher())
                .bookCondition(rent.getBookCondition())
                .bookImage(rent.getBookImage())
                .address(rent.getAddress())
                .rentStatus(rent.getRentStatus().name())
                .borrowerNickname(borrowerNickname)
                .createdDate(rent.getCreatedDate())
                .modifiedDate(rent.getModifiedDate())
                .returnDate(returnDate)
                .hasReview(hasReview)
                .build();
    }
}

File: backend/src/main/java/com/bookbook/domain/lendList/dto/LendListResponseDto.kt
package com.bookbook.domain.lendList.dto

import com.bookbook.domain.rent.entity.Rent
import java.time.LocalDateTime

data class LendListResponseDto(
    val id: Long,
    val lenderUserId: Long?,
    val title: String?,
    val bookTitle: String?,
    val author: String?,
    val publisher: String?,
    val bookCondition: String?,
    val bookImage: String?,
    val address: String?,
    val rentStatus: String?,
    val borrowerNickname: String?,
    val createdDate: LocalDateTime,
    val modifiedDate: LocalDateTime,
    val returnDate: LocalDateTime?,
    val hasReview: Boolean
) {
    constructor(
        rent: Rent,
        borrowerNickname: String?,
        returnDate: LocalDateTime?,
        hasReview: Boolean
    ) : this(
        id = rent.id,
        lenderUserId = rent.lenderUserId,
        title = rent.title,
        bookTitle = rent.bookTitle,
        author = rent.author,
        publisher = rent.publisher,
        bookCondition = rent.bookCondition,
        bookImage = rent.bookImage,
        address = rent.address,
        rentStatus = rent.rentStatus?.name,
        borrowerNickname = borrowerNickname,
        createdDate = rent.createdDate,
        modifiedDate = rent.modifiedDate,
        returnDate = returnDate,
        hasReview = hasReview
    )
}